modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue

```diff
@@ -1,93 +1,140 @@
+                @keydown.prevent.arrow-down="switchGroup(itemIndex, 1)"
+                @keydown.prevent.arrow-up="switchGroup(itemIndex, -1)"
+                @keydown.prevent.home="switchGroup(itemIndex, 0)"
+                @keydown.prevent.end="switchGroup(itemIndex, null)"
                 ref="groupButton"
               >
                 <span>{{ item.label }}</span>
                 <chevron-up-icon
                   class="apos-area-menu__group-chevron"
-                  :class="{'is-active': index === active}" :size="13"
+                  :class="{'is-active': itemIndex === active}" :size="13"
                 />
               </button>
             </dt>
             <dd class="apos-area-menu__group-list" role="region">
               <ul
                 class="apos-area-menu__items apos-area-menu__items--accordion"
-                :class="{'is-active': active === index}"
-                :id="`${menuId}-group-${index}`"
-                :aria-labelledby="`${menuId}-trigger-${index}`"
-                :aria-expanded="active === index ? 'true' : 'false'"
+                :class="{'is-active': active === itemIndex}"
+                :id="`${menuId}-group-${itemIndex}`"
+                :aria-labelledby="`${menuId}-trigger-${itemIndex}`"
+                :aria-expanded="active === itemIndex ? 'true' : 'false'"
               >
                 <li
                   class="apos-area-menu__item"
                   v-for="(child, childIndex) in item.items"
-                  :key="child.action"
+                  :key="child.name"
                   :ref="`child-${index}-${childIndex}`"
                 >
                   <AposAreaMenuItem
-                    @click="clicked(child)"
+                    @click="add(child.name)"
                     :item="child"
-                    :tabbable="index === active"
-                    @up="switchItem(`child-${index}-${childIndex - 1}`, -1)"
-                    @down="switchItem(`child-${index}-${childIndex + 1}`, 1)"
+                    :tabbable="itemIndex === active"
+                    @up="switchItem(`child-${itemIndex}-${childIndex - 1}`, -1)"
+                    @down="switchItem(`child-${itemIndex}-${childIndex + 1}`, 1)"
                   />
                 </li>
               </ul>
             </dd>
           </dl>
           <AposAreaMenuItem
             v-else
-            @click="clicked(item)"
+            @click="add(item.name)"
             :item="item"
-            @up="switchItem(`item-${index - 1}`, -1)"
-            @down="switchItem(`item-${index + 1}`, 1)"
+            @up="switchItem(`item-${itemIndex - 1}`, -1)"
+            @down="switchItem(`item-${itemIndex + 1}`, 1)"
+            @close="close"
           />
         </li>
       </ul>
     </AposContextMenu>
+    <component
+      :is="addWidgetEditor"
+      v-if="adding"
+      v-model="widget"
+      :type="addWidgetType"
+      @close="close()"
```

Comment:
This doesn't need to be invoked like this. Simply `@close="close"`.

Reply:
removed